packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -138,6 +139,24 @@ class OverlayEntry implements Listenable {
     _overlay!._didChangeEntryOpacity();
   }
 
+  /// Whether the content of this [OverlayEntry] can be used to size the
+  /// [Overlay].
+  ///
+  /// In most situations the Overlay sizes itself based on its incoming
+  /// constraints to be as large as possible. However, if that would result in
+  /// an infinite size, it has to rely on one of its children to size itself. In
+  /// this situation, the Overlay will consult the topmost [OverlayEntry] that
+  /// has this property set to true, lay it out with unconstrained
```

Comment:
nit: not necessarily completely unconstrained I think? Maybe it's just infinite maxWidth or infinite maxHeight or infinite both? I feel `unconstrained [BoxConstraints]` sounded like we are going to throw the incoming constraints away.

Reply:
(Can a dynamically sized view give you constrained width but unconstrained height?)

Reply:
> I feel unconstrained [BoxConstraints] sounded like we are going to throw the incoming constraints away.

Oh, good catch. I'll update the docs to make it clear that we just pass on the incoming constraints in that case.

> (Can a dynamically sized view give you constrained width but unconstrained height?)

Yes, this is allowed.